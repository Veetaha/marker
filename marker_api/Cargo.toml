[package]
name = "marker_api"

categories  = ["development-tools"]
description = "Marker's API, designed for stability and usability"

edition    = { workspace = true }
keywords   = { workspace = true }
license    = { workspace = true }
repository = { workspace = true }
version    = { workspace = true }

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
visibility = { workspace = true }

[dev-dependencies]
expect-test = { workspace = true }

[features]
# Some items should only be used by the driver implementing the functionality,
# this feature enables the export of these items. Note that this interface is
# unstable. All usage of the driver api can change between releases.
driver-api = []
