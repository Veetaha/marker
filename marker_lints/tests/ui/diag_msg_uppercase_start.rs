extern crate marker_api;

use marker_api::{ast::expr::ExprKind, context::AstContext};

marker_api::declare_lint!{
    /// Dummy
    DUMMY,
    Warn,
}

pub fn accept_message<'ast>(cx: &AstContext<'ast>, expr: ExprKind<'ast>) {
    cx.emit_lint(DUMMY, expr, "x <-- this is cool");
    cx.emit_lint(DUMMY, expr, "=^.^= <-- Interesting, but valid");
    cx.emit_lint(DUMMY, expr, "");

    let variable = "";
    cx.emit_lint(DUMMY, expr, variable);
}

pub fn warn_about_message<'ast>(cx: &AstContext<'ast>, expr: ExprKind<'ast>) {
    cx.emit_lint(DUMMY, expr, "X <-- starting with upper case");
    cx.emit_lint(DUMMY, expr, "Hey <-- starting with upper case");
}

fn main() {}
