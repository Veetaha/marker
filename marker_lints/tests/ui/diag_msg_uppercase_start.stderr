error[E0603]: module `expr` is private
 --> $DIR/diag_msg_uppercase_start.rs:3:23
  |
3 | use marker_api::{ast::expr::ExprKind, context::MarkerContext};
  |                       ^^^^ private module
  |
note: the module `expr` is defined here
 --> /home/xfrednet/workspace/rust/marker/marker_api/src/ast.rs:5:1
  |
5 | mod expr;
  | ^^^^^^^^
help: consider importing this enum instead
  |
3 | use marker_api::{marker_api::prelude::ExprKind, context::MarkerContext};
  |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: aborting due to previous error

For more information about this error, try `rustc --explain E0603`.
